### ISSUE TITLE
Makefile appears broken - touching header causes no build

### ISSUE THREAD

**Issue by vsajip** [python/cpython#82445]
BPO | [38264](https://bugs.python.org/issue38264)
--- | :---
Nosy | @vsajip

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2019-09-24.17:56:57.466>
labels = ['3.8', 'type-bug', '3.7', '3.9', 'build']
title = 'Makefile appears broken - touching header causes no build'
updated_at = <Date 2019-09-24.17:58:39.383>
user = 'https://github.com/vsajip'
```

bugs.python.org fields:
```python
activity = <Date 2019-09-24.17:58:39.383>
actor = 'vinay.sajip'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Build']
creation = <Date 2019-09-24.17:56:57.466>
creator = 'vinay.sajip'
dependencies = []
files = []
hgrepos = []
issue_num = 38264
keywords = []
message_count = 2.0
messages = ['353105', '353106']
nosy_count = 1.0
nosy_names = ['vinay.sajip']
pr_nums = []
priority = 'high'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue38264'
versions = ['Python 3.7', 'Python 3.8', 'Python 3.9']
```

</p></details>

**Comment by crazymerlyn** [comment]
Should this be closed? Looks like a duplicate of #89758 which has been fixed. I can't reproduce it on HEAD.

**Comment by itamaro** [comment]
closing. please reopen (or submit a new issue) if the issue can be reproduced on main branch.
